core/services/keeper/upkeep_executer.go

```diff
@@ -313,22 +310,37 @@ func buildJobSpec(
 	gasPrice *big.Int,
 	gasTipCap *big.Int,
 	gasFeeCap *big.Int,
-	chainID string,
+	chainId *utils.Big,
 ) map[string]interface{} {
+	evmChainID := ""
+	if chainId != nil {
+		evmChainID = chainId.String()
+	}
+
+	var checkUpkeepGasLimit, performUpkeepGasLimit uint32
+	if !isArbitrum(chainId) {
+		performUpkeepGasLimit = upkeep.ExecuteGas + ormConfig.KeeperRegistryPerformGasOverhead()
+		checkUpkeepGasLimit = exConfig.KeeperRegistryCheckGasOverhead() + upkeep.Registry.CheckGas +
+			exConfig.KeeperRegistryPerformGasOverhead() + upkeep.ExecuteGas
+	}
+
 	return map[string]interface{}{
 		"jobSpec": map[string]interface{}{
 			"jobID":                 jb.ID,
 			"fromAddress":           upkeep.Registry.FromAddress.String(),
 			"contractAddress":       upkeep.Registry.ContractAddress.String(),
 			"upkeepID":              upkeep.UpkeepID.String(),
 			"prettyID":              upkeep.PrettyID(),
-			"performUpkeepGasLimit": upkeep.ExecuteGas + ormConfig.KeeperRegistryPerformGasOverhead(),
-			"checkUpkeepGasLimit": exConfig.KeeperRegistryCheckGasOverhead() + upkeep.Registry.CheckGas +
-				exConfig.KeeperRegistryPerformGasOverhead() + upkeep.ExecuteGas,
-			"gasPrice":   gasPrice,
-			"gasTipCap":  gasTipCap,
-			"gasFeeCap":  gasFeeCap,
-			"evmChainID": chainID,
+			"performUpkeepGasLimit": performUpkeepGasLimit,
+			"checkUpkeepGasLimit":   checkUpkeepGasLimit,
+			"gasPrice":              gasPrice,
+			"gasTipCap":             gasTipCap,
+			"gasFeeCap":             gasFeeCap,
+			"evmChainID":            evmChainID,
 		},
 	}
 }
+
+func isArbitrum(chainId *utils.Big) bool {
+	return chainId != nil && (chainId.Int64() == arbitrumChainId || chainId.Int64() == arbitrumGoerli)
+}
```

Comment:
We shouldn't be hard coding chain id logic anywhere in here. It should be possible to leverage `ChainType==Arbitrum` instead, but IMHO that should only be a last resort. If this is functional with unlimited estimations, and we cannot justify why these limits are set in the first place, then we should just drop them for all chains. Otherwise we are just over constraining ourselves and should favor flexibility when it doesn't hurt correctness.